### ISSUE TITLE
Bug AST changed error

### ISSUE THREAD

**Issue by Pitometsu** [ocaml-ppx/ocamlformat#936]
```ocaml
let () = match a with
         | `A-> b c d >>= (function
               | `B -> ()
               | `C -> ())
         | `D -> failwith ""
```

**Comment by gpetiot** [comment]
I tested this with the default, conventional, janestreet, compact and sparse profiles, and did not get any error. What options did you set?

**Comment by Pitometsu** [comment]
.ocamlformat
```
single-case=compact
break-separators=before
break-sequences=true
break-cases=fit
break-collection-expressions=fit-or-vertical
break-infix=fit-or-vertical
module-item-spacing=compact
doc-comments=before
ocp-indent-compat=true
type-decl=sparse
let-and=sparse
field-space=loose
extension-sugar=always
max-iters=3
quiet=true
cases-exp-indent=4
break-infix-before-func=false
leading-nested-match-parens=true
indicate-multiline-delimiters=false
indicate-nested-or-patterns=false
space-around-collection-expressions=false
```
.ocp-indent
```
max_indent=2
align_ops=false
align_params=false
match_clause=4
```
version from 55fe414ad844ae9847a9722bfaa6ae471dc277e3 (latest working for OCaml `4.05`).

**Comment by gpetiot** [comment]
Fixed with #760
